### PR TITLE
Try to document release conventions.

### PR DESCRIPTION
**Pull request by oponder**
From a talk I had with marcel when he tried to explain me the situation.

I tried to note it down in a way that might help others and customers.

fyi @MarcelMue

### CODE REVIEW

File: src/content/reference/release-versions/_index.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+---
+title: Release Versions
+description: Here you find our reference regarding our release versions
+date: 2020-05-14
+layout: subsection
+weight: 500
+---
+
+# Giant Swarm Releases Versions
```
Comment (marians) [inline src/content/reference/release-versions/_index.md:9]:
```suggestion
# Giant Swarm Release Versions
```

**Comment by marians** [comment]
This seems to overlap with https://docs.giantswarm.io/reference/cluster-upgrades/

I am in favour of having a page dedicated to the release concept. The cluster upgrades page should be modified to link to this page then.

File: src/content/reference/release-versions/_index.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+# Giant Swarm Release Versions
+
+When you create a tenant cluster at Giant Swarm you also get to choose which
+release version you would like the cluster to have.
+
+The release version of a cluster helps you identify additional information like
+what Kubernetes version the cluster will have.
+
+Besdies Kubernetes, other components come into play as well, for example Calico,
```
Comment (oponder) [inline src/content/reference/release-versions/_index.md:17]:
```suggestion
Besides Kubernetes, other components come into play as well, for example Calico,
```

**Comment by oponder** [comment]
@marians I added a link from the cluster upgrade reference page to this page.

Are there any other content change you wanted on this page? I think they can co-exist.

File: src/content/reference/cluster-upgrades/_index.md [hunk 3]
```diff
@@ -62,6 +62,9 @@ This can include minor upgrades of third party components, with the exception of
 **Note:** The release version number is provider-specific.
 Azure, AWS, and KVM based installations have independent versioning systems, as their stacks are also slightly different.
 
+We have a [reference page with an overview of our release versions](/reference/release-versions/), which can give you an idea of what
+release versions are currently available and what notable features they support.
+
```
Comment (pipo02mix) [inline src/content/reference/cluster-upgrades/_index.md:67]:
```suggestion
We have a [reference page with an overview of our release versions](/reference/release-versions/), which can give you an idea of what release versions are currently available and what notable features they support.
```

File: src/content/reference/release-versions/_index.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+we create a new release version so that you can upgrade your existing tenant clusters
+and create new clusters with the added or fixed functionality.
+
+## Conventions around release versioning
+
+We've had to break these conventions in some cases, but we're trying to re-align
+things as time goes on and customers no longer use older versions.
+
+### Major versions increment on Kubernetes releases
+We generally try to increment our Major version only when there is a new Kubernetes release.
```
Comment (pipo02mix) [inline src/content/reference/release-versions/_index.md:32]: Or a breaking change, right?

Comment (oponder) [reply src/content/reference/release-versions/_index.md:32]: Not sure... so far we've had breaking changes across major version releases as well as in patch release

**Comment by oponder** [comment]
@pipo02mix @marians LGTY?

About the two goals to add, can we add them as another PR? For now I'd like to get this out if possible and then we can think about how best to word the things you mentioned.

**Comment by oponder** [comment]
I also added information about the App Platform and when it got enabled.

File: src/content/reference/release-versions/_index.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+|**v13.x.x** | Kubernetes 1.18.x | not yet available|
+
+
+#### Exceptions
+
+On AWS, **v9.2.0** and up contains **Kubernetes 1.16.x**
+
+## Versions that support node pools
+
+Currently only AWS **v11.x.x** and up support nodepools.
```
Comment (cokiengchiara) [inline src/content/reference/release-versions/_index.md:49]: @oponder is it true? I thought Node Pools was introduced in 10.x.x https://github.com/giantswarm/releases/blob/master/release-notes/aws/v10.1.0.md

Comment (oponder) [reply src/content/reference/release-versions/_index.md:49]: correct, but there are no active 10.x.x releases now (i think)

**Pull request merged**